target_chains/ethereum/contracts/forge-test/Entropy.t.sol

```diff
@@ -530,4 +548,76 @@ contract EntropyTest is Test {
         assertEq(providerInfo1.uri, provider1Uri);
         assertEq(providerInfo1.commitmentMetadata, provider1CommitmentMetadata);
     }
+
+    function testSetAdminByAdmin() public {
+        vm.prank(admin);
+        random.setAdmin(admin2);
+        assertEq(random.getAdmin(), admin2);
+    }
+
+    function testSetAdminByOwner() public {
+        vm.prank(owner);
+        random.setAdmin(admin2);
+        assertEq(random.getAdmin(), admin2);
+    }
+
+    function testFailSetAdminByUnauthorised() public {
+        vm.prank(admin2);
+        random.setAdmin(admin);
```

Comment:
I think `testFail` is deprecated. I think the right way to do this is to use `vm.expectRevert()` before the call that you expect to fail.